pkg/sql/rowexec/project_set.go

```diff
@@ -125,7 +125,15 @@ func newProjectSetProcessor(
 // MustBeStreaming implements the execinfra.Processor interface.
 func (ps *projectSetProcessor) MustBeStreaming() bool {
 	// If we have a single streaming generator, then the processor is such too.
-	for _, gen := range ps.gens {
+	for _, fn := range ps.funcs {
+		if fn == nil {
+			continue
+		}
+		gen, err := ps.generatorForFuncExpr(fn)
+		if err != nil {
+			continue
+		}
+		defer gen.Close(ps.Ctx)
```

Comment:
Unfortunately, this Close() might assume that Start() has run. Calling Start() here would be an even bigger bummer.

So, I think this isn't how we want to initialize these.